packages/components/src/box-control/all-input-control.tsx

```diff
@@ -69,18 +83,43 @@ export default function AllInputControl( {
 		} );
 	};
 
+	const [ parsedQuantity, parsedUnit ] =
+		parseQuantityAndUnitFromRawValue( allValue );
+
 	return (
-		<UnitControl
-			{ ...props }
-			disableUnits={ isMixed }
-			isOnly
-			value={ allValue }
-			onChange={ handleOnChange }
-			onUnitChange={ handleOnUnitChange }
-			onFocus={ handleOnFocus }
-			onHoverOn={ handleOnHoverOn }
-			onHoverOff={ handleOnHoverOff }
-			placeholder={ allPlaceholder }
-		/>
+		<HStack>
+			<UnitControl
+				{ ...props }
+				disableUnits={ isMixed }
+				id={ inputId }
+				isOnly
+				value={ parsedQuantity }
+				onChange={ handleOnChange }
+				onUnitChange={ handleOnUnitChange }
+				onFocus={ handleOnFocus }
+				onHoverOn={ handleOnHoverOn }
+				onHoverOff={ handleOnHoverOff }
+				placeholder={ allPlaceholder }
+			/>
+
+			<FlexedRangeControl
+				__nextHasNoMarginBottom
+				aria-controls={ inputId }
+				aria-labelledby={ inputId }
+				hideLabelFromVision
+				initialPosition={ parsedQuantity ?? 0 }
+				label={ LABELS.all }
+				onChange={ ( newValue ) => {
+					sliderOnChange( [ newValue, parsedUnit ].join( '' ) );
```

Comment:
We could refactor the code slightly and resuse as much as possible of the existing logic, which not only makes the code less redundant, but also ensures that change events on `RangeControl` would have the exact same effect as the ones on `UnitControl`

<details>

<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/box-control/all-input-control.tsx b/packages/components/src/box-control/all-input-control.tsx
index a6aefab80d..e1f09a16a3 100644
--- a/packages/components/src/box-control/all-input-control.tsx
+++ b/packages/components/src/box-control/all-input-control.tsx
@@ -39,18 +39,16 @@ export default function AllInputControl( {
 	const isMixed = hasValues && isValuesMixed( values, selectedUnits, sides );
 	const allPlaceholder = isMixed ? LABELS.mixed : undefined;
 
+	const [ parsedQuantity, parsedUnit ] =
+		parseQuantityAndUnitFromRawValue( allValue );
+
 	const handleOnFocus: React.FocusEventHandler< HTMLInputElement > = (
 		event
 	) => {
 		onFocus( event, { side: 'all' } );
 	};
 
-	const sliderOnChange = ( next: string ) => {
-		const nextValues = applyValueToSides( values, String( next ), sides );
-		onChange( nextValues );
-	};
-
-	const handleOnChange: UnitControlProps[ 'onChange' ] = ( next ) => {
+	const onValueChange = ( next?: string ) => {
 		const isNumeric = next !== undefined && ! isNaN( parseFloat( next ) );
 		const nextValue = isNumeric ? next : undefined;
 		const nextValues = applyValueToSides( values, nextValue, sides );
@@ -58,6 +56,16 @@ export default function AllInputControl( {
 		onChange( nextValues );
 	};
 
+	const unitControlOnChange: UnitControlProps[ 'onChange' ] = ( next ) => {
+		onValueChange( next );
+	};
+
+	const sliderOnChange = ( next?: number ) => {
+		onValueChange(
+			next !== undefined ? [ next, parsedUnit ].join( '' ) : undefined
+		);
+	};
+
 	// Set selected unit so it can be used as fallback by unlinked controls
 	// when individual sides do not have a value containing a unit.
 	const handleOnUnitChange: UnitControlProps[ 'onUnitChange' ] = ( unit ) => {
@@ -83,9 +91,6 @@ export default function AllInputControl( {
 		} );
 	};
 
-	const [ parsedQuantity, parsedUnit ] =
-		parseQuantityAndUnitFromRawValue( allValue );
-
 	return (
 		<HStack>
 			<UnitControl
@@ -94,7 +99,7 @@ export default function AllInputControl( {
 				id={ inputId }
 				isOnly
 				value={ parsedQuantity }
-				onChange={ handleOnChange }
+				onChange={ unitControlOnChange }
 				onUnitChange={ handleOnUnitChange }
 				onFocus={ handleOnFocus }
 				onHoverOn={ handleOnHoverOn }
@@ -109,9 +114,7 @@ export default function AllInputControl( {
 				hideLabelFromVision
 				initialPosition={ parsedQuantity ?? 0 }
 				label={ LABELS.all }
-				onChange={ ( newValue ) => {
-					sliderOnChange( [ newValue, parsedUnit ].join( '' ) );
-				} }
+				onChange={ sliderOnChange }
 				min={ 0 }
 				max={ CUSTOM_VALUE_SETTINGS[ parsedUnit ?? 'px' ]?.max ?? 10 }
 				step={

```
</details>